pkg/applications/applications.go

```diff
@@ -0,0 +1,117 @@
+package applications
+
+import (
+	v1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	"github.com/jenkins-x/jx/pkg/cmd/clients"
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/kube/services"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	"k8s.io/api/apps/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type applicationDeployment struct {
+	Deployment *v1beta1.Deployment
+}
+
+type applicationEnvironment struct {
+	Environment v1.Environment
+	Deployments []applicationDeployment
+}
+
+// Application represents an application created by the user
+type Application struct {
+	*v1.SourceRepository
+	Environments map[string]applicationEnvironment
+}
+
+// Name returns the name of an application
+func (a Application) Name() string {
+	return a.SourceRepository.Spec.Repo
+}
+
+// Version returns the version of a deployment
+func (ad applicationDeployment) Version() string {
+	return kube.GetVersion(&ad.Deployment.ObjectMeta)
+}
+
+// Pods returns the ratio of pods that are ready/replicas
+func (ad applicationDeployment) Pods() string {
```

Comment:
Is that needed somewhere for displaying?

Reply:
yes, we display it on the table view unless the user specifies they don't want it